### PR TITLE
Layer dragging and dropping

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Part of https://github.com/CartoDB/cartodb/issues/7342.

drag droppin

@javierarce 👍  or 👎  ? thx!

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js [hunk 1]
```diff
@@ -11,6 +12,69 @@ var layerTypesAndKinds = require('./layer-types-and-kinds');
...
+  },
+
+  _resetOrders: function () {
+    this.each(function (layerDefModel, index) {
+      layerDefModel.set('order', index, { silent: true });
+    });
+  },
+
+  save: function (options) {
+    var promises = this.chain()
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: @viddo FYI: this is how I'm making sure analysis definition nodes are created before the collection of layers is saved (now triggered after user re-orders some layers).

I'll open a separate issue to discuss if we should do this in bulk and if we should make this operation atomic.

Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/layer-definitions-collection.js]: 👌

File: app/assets/stylesheets/editor-3/layers-list.css.scss [hunk 2]
```diff
@@ -14,6 +14,9 @@
   background: #FFF;
   position:relative;
 }
+.Editor-ListLayer-item--placeholder {
+  background: #F2F6F9;
```
Comment (javierarce) [inline app/assets/stylesheets/editor-3/layers-list.css.scss]: This color is almost white, right? Not sure if this is right or we could reduce a bit the opacity.

**Comment by javierarce** [comment]
👍  (I added just a super minor comment, though)

File: app/assets/stylesheets/editor-3/layers-list.css.scss (hunk 2 above)
Comment (alonsogarciapablo) [inline app/assets/stylesheets/editor-3/layers-list.css.scss]: I used a color picker to pick the color we have in the mockups. Not sure about this. What color would you recommend? Here's the actual color:

screen shot 2016-06-01 at 11 34 49

Comment (alonsogarciapablo) [inline app/assets/stylesheets/editor-3/layers-list.css.scss]: Asking @piensaenpixel about this.

Comment (javierarce) [inline app/assets/stylesheets/editor-3/layers-list.css.scss]: Oh, that color is for the placeholder! I thought it was for a different thing, sorry. Then I guess it's fine.

Comment (alonsogarciapablo) [inline app/assets/stylesheets/editor-3/layers-list.css.scss]: I changed a bit anyway in https://github.com/CartoDB/cartodb/pull/7581/commits/4fa353ae6541798a12db82b810ae5fc3329678da. I replaced color by existing variable and changed the color of the border to match the background color. Thanks @piensaenpixel!

**Comment by alonsogarciapablo** [comment]
Thanks for the CR @javierarce! Merging after all tests are green.

**Pull request merged**